axum/src/lib.rs

```diff
@@ -260,6 +268,68 @@
 //! # };
 //! ```
 //!
+//! The downside to this approach is that you'll get runtime errors
+//! (specifically a `500 Internal Server Error` response) if you try and extract
+//! an extension that doesn't exist, perhaps because you forgot add the
+//! middleware or because you're extracting the wrong type.
```

Comment:
Closure captures should also have lower runtime cost as the clones only happen when that specific route is invoked, right?

Reply:
Not sure whether it's worth to point that out in docs though.